### ISSUE TITLE
[Clang] Clang assertion failure when run with `-cc1 -ast-dump` args and a LLVM IR file

### ISSUE THREAD

**Issue by yronglin** [llvm/llvm-project#88522]
Clang assertion failed at https://github.com/llvm/llvm-project/blob/599adf30afe5802fab80419ec5bb896036a1c8fb/clang/lib/Frontend/FrontendAction.cpp#L759-L761 , I think we'de better to add a diagnostic message instead assertion failure?
```
Assertion failed: (hasIRSupport() && "This action does not have IR file support!"), function BeginSourceFile, file FrontendAction.cpp, line 761.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
```

**Comment by shafik** [comment]
It does seem like this could be a diagnostic as opposed to a assertion, it was landed a long time ago: 9507f9cc054da

**Comment by yronglin** [comment]
> It does seem like this could be a diagnostic as opposed to a assertion, it was landed a long time ago: [9507f9c](https://github.com/llvm/llvm-project/commit/9507f9cc054da3b8148ac294ea5ced79625a897f)

Thanks for confirmation! I've a submitted PR to add a diagnostic.
